autokeras/auto/auto_model.py

```diff
@@ -214,9 +152,107 @@ def _add_node(self, input_node):
+            hypermodel=self,
+            objective='val_loss',
+            max_trials=self.trials,
+            directory=self.directory)
+
+        # TODO: allow early stop if epochs is not specified.
+        self.tuner.search(x=x,
+                          y=y,
+                          validation_data=validation_data,
+                          **kwargs)
+
+    def predict(self, x, **kwargs):
+        """Predict the output for a given testing data. """
+        return self.tuner.get_best_models(1)[0].predict(x, **kwargs)
+
+    def _get_metrics(self):
+        metrics = []
+        for metrics_list in [output_node.in_hypermodels[0].metrics for
+                             output_node in self.outputs
+                             if isinstance(output_node.in_hypermodels[0],
+                                           hyper_head.HyperHead)]:
+            metrics += metrics_list
+        return metrics
+
     def _get_loss(self):
         loss = nest.flatten([output_node.in_hypermodels[0].loss
                              for output_node in self.outputs
                              if isinstance(output_node.in_hypermodels[0],
                                            hyper_head.HyperHead)])
         return loss
+
+    def _compiled(self, hp, model):
+        # Specify hyperparameters from compile(...)
+        optimizer = hp.Choice('optimizer',
+                              ['adam',
+                               'adadelta',
+                               'sgd'])
+
+        model.compile(optimizer=optimizer,
+                      metrics=self._get_metrics(),
+                      loss=self._get_loss())
+
+        return model
+
+
+class AutoModel(GraphAutoModel):
+    """ A HyperModel defined by inputs and outputs.
+
+    AutoModel is a subclass of HyperModel. Besides the HyperModel properties,
+    it also has a tuner to tune the HyperModel. The user can use it in a similar
+    way to a Keras model since it also has `fit()` and  `predict()` methods.
+
+    The user can specify the inputs and outputs of the AutoModel. It will infer
+    the rest of the high-level neural architecture.
+
+    Attributes:
+        inputs: A list of or a HyperNode instance.
+            The input node(s) of a the AutoModel.
+        outputs: A list of or a HyperHead instance.
+            The output head(s) of the AutoModel.
+        trials: Int. The number of models to try.
+        directory: String. The path to a directory for storing the search outputs.
+    """
+
+    def __init__(self,
+                 inputs,
+                 outputs,
+                 **kwargs):
+        inputs = layer_utils.format_inputs(inputs)
+        outputs = layer_utils.format_inputs(outputs)
+        middle_nodes = [input_node.related_block()(input_node)
+                        for input_node in inputs]
+        if len(middle_nodes) > 1:
+            output_node = hyper_block.Merge()(middle_nodes)
+        else:
+            output_node = middle_nodes[0]
```

Comment:
Why do we need this? Can't you let `hyper_block.Merge()` taking care of the merge for a list that contains one node?